hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/XceiverServerSpi.java

```diff
@@ -83,4 +85,12 @@ default void removeGroup(HddsProtos.PipelineID pipelineId)
    * @return list of report for each pipeline.
    */
   List<PipelineReport> getPipelineReport();
+
+  /**
+   * Get storage report for the XceiverServer instance.
+   * @return list of report for each storage location.
+   */
+  default List<MetadataStorageReportProto> getStorageReport() throws IOException {
```

Comment:
It seems to be a good time the remove XceiverServerSpi interface. Originally we had two separated implementation to connect to the datanode. Today we have only one. One interface is used between the client and the datanode, and the other one between datanode and ratis (datanode). As this example shows, the two interface shouldn't be the same.

Reply:
Definitely, yes..
Filed https://issues.apache.org/jira/browse/HDDS-3963 for the same